src/store/modules/farmer/types.ts

```diff
@@ -0,0 +1,41 @@
+import { CoreState, CoreData, PartialCoreData } from '../../types';
+
+/*----------------------- Models -----------------------*/
+
+interface BaseFarmer {
+  username: string;
+  firstName: string;
+  middleName: string;
+  lastName: string;
+  phoneCountry?: string;
+  phoneArea?: string;
+  phoneNumber?: string;
+  companyName?: string;
+}
+
+/** Model for Farmer */
+export type Farmer = CoreData<BaseFarmer>;
+
+/** Model for updating farmer */
+export type PartialFarmer = PartialCoreData<BaseFarmer>;
+
+/*----------------------- State -----------------------*/
+
+/** Farmer module state */
+export interface FarmerState extends CoreState {
+  farmers: Farmer[];
+}
+
+/*----------------------- Actions -----------------------*/
+
+/** Actions for farmer module */
+export type Action = {
+  type: 'UPDATE_FARMER',
+  farmer: PartialFarmer,
+} | {
+  type: 'CREATE_FARMER',
+  farmer: Farmer,
+} | {
+  type: 'DO_NOT_USE',
```

Comment:
What's this for?

Reply:
Annoyingly we can't call our `Action` type an extension of the redux `Action` unless we have one type that has no payload. So for modules w/ no payload we have to add this. Should be harmless, it won't do anything because we don't cover it in the reducer

Reply:
Oh ok, weird.